sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuListUtils.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+import java.util.Optional
+
+import ai.rapids.cudf.{ColumnView, DType}
+
+/**
+ * Provide a set of APIs to manipulate array/list columns in common ways.
+ */
+object GpuListUtils extends Arm {
+  /**
+   * Replace the data column in a LIST column. This will keep the same offsets and validity
+   * of the listColumn.  This returns a view so it is the responsibility of the caller to keep
+   * both listCol and newDataCol alive longer than the returned ColumnView.
+   * @param listCol the list column to use as a template
+   * @param newDataCol the new data column.
+   * @return a new ColumnView.
+   * @throws IllegalArgumentException if data column does not match the original data column in
+   *                                  size.
+   */
+  def replaceListDataColumn(
```

Comment:
Given this is intended to be a general utility, should this have something like "asView" in the name to help drive home the dangerousness of this method?  If someone assumes this creates a column instead of a column view it could get ugly to debug.  Or are you convinced that other APIs that require a ColumnVector instead of a ColumnView would prevent this mistake in practice?